test/org/javarosa/model/xform/SMSSerializingVisitorTest.java

```diff
@@ -0,0 +1,161 @@
+package org.javarosa.model.xform;
+
+import org.javarosa.core.PathConst;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.services.transport.payload.ByteArrayPayload;
+import org.javarosa.core.test.FormParseInit;
+import org.javarosa.form.api.FormEntryController;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.javarosa.test.utils.Utils.convertToString;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Performs a comprehensive suite of tests on the results of a form
+ * being serialized to the SMS format.
+ */
+public class SMSSerializingVisitorTest {
+    private FormParseInit formParser;
+    private String SMS;
+    private static final String delimiter = " ";
+    private FormEntryController formEntryController;
+    private FormInstance formInstance;
```

Comment:
Following these IDEA suggestions can simplify the code.

![screen shot 2018-06-18 at 11 45 39 am](https://user-images.githubusercontent.com/55099/41555658-3954d7f0-72ed-11e8-9cd3-1089d35c3860.png)